internal/service/ipam/api_fixed_address_resource_test.go

```diff
@@ -0,0 +1,545 @@
+	var v ipam.IpamsvcFixedAddress
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccFixedAddressTags("10.0.0.10", "mac", "aa:aa:aa:aa:aa:aa", map[string]string{
+					"tag1": "value1",
+					"tag2": "value2",
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFixedAddressExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag1", "value1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag2", "value2"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccFixedAddressTags("10.0.0.10", "mac", "aa:aa:aa:aa:aa:aa", map[string]string{
+					"tag2": "value2changed",
+					"tag3": "value3",
+				}),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckFixedAddressExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag2", "value2changed"),
+					resource.TestCheckResourceAttr(resourceName, "tags.tag3", "value3"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func testAccCheckFixedAddressExists(ctx context.Context, resourceName string, v *ipam.IpamsvcFixedAddress) resource.TestCheckFunc {
+	// Verify the resource exists in the cloud
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		apiRes, _, err := acctest.BloxOneClient.IPAddressManagementAPI.
+			FixedAddressAPI.
+			FixedAddressRead(ctx, rs.Primary.ID).
+			Execute()
+		if err != nil {
+			return err
+		}
+		if !apiRes.HasResult() {
+			return fmt.Errorf("expected result to be returned: %s", resourceName)
+		}
+		*v = apiRes.GetResult()
+		return nil
+	}
+}
+
+func testAccCheckFixedAddressDestroy(ctx context.Context, v *ipam.IpamsvcFixedAddress) resource.TestCheckFunc {
+	// Verify the resource was destroyed
+	return func(state *terraform.State) error {
+		_, httpRes, err := acctest.BloxOneClient.IPAddressManagementAPI.
+			FixedAddressAPI.
+			FixedAddressRead(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			if httpRes != nil && httpRes.StatusCode == http.StatusNotFound {
+				// resource was deleted
+				return nil
+			}
+			return err
+		}
+		return errors.New("expected to be deleted")
+	}
+}
+
+func testAccCheckFixedAddressDisappears(ctx context.Context, v *ipam.IpamsvcFixedAddress) resource.TestCheckFunc {
+	// Delete the resource externally to verify disappears test
+	return func(state *terraform.State) error {
+		_, err := acctest.BloxOneClient.IPAddressManagementAPI.
+			FixedAddressAPI.
+			FixedAddressDelete(ctx, *v.Id).
+			Execute()
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func testAccBaseWithIPSpaceAndSubnetAndRange() string {
```

Comment:
do you really need range for FA, maybe you can use `testAccBaseWithIPSpaceAndSubnet`

Reply:
FA can be created under subnet or under ranges, for UT we can skip the range part. will update the PR.